### PR TITLE
#120: API-core-initialize_finalize-finalize from md to rst

### PR DESCRIPTION
**Pull request by pierrepebay**
### **THIS PR:**
* API/core/initialize_finalize/finalize transition from .md to .rst

before (.md) | after (.rst)
:----------------: | :-------------: 
before | after

### CODE REVIEW

**Comment by dalg24** [comment]
To make sure I understand: The only observable difference is the `finalize` header but that is presumably intentional since the header is in code font in the .rst file.

**Comment by fnrizzi** [comment]
@dalg24 correct, because this file is simple. The major differences will be mostly for files containing cpp classes that will need to be converted to the Sphinx cpp class domain/syntax. But converting everything to rst allows us to setup everything so that if later we need to change/add things, it will be already ready to use Sphinx

**Comment by fnrizzi** [comment]
And if for example we want to customize the rendering of a cpp function via 
```
cppkokkos::function
```
We have everything ready to support that

**Comment by fnrizzi** [comment]
I just did above , we overlapped

**Comment by dalg24** [comment]
OK I commented before I saw Francesco's reply.  I assume the rational is "so that we can use the customized rendering everywhere".

File: docs/source/API/core/initialize_finalize/finalize.rst [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+Once this function is called, no Kokkos API functions (not even
+`Kokkos::initialize <initialize.html#kokkosinitialize>`_) may be called, except for
+``Kokkos::is_initialized`` or ``Kokkos::is_finalized``.
+The user must ensure that all Kokkos objects (e.g. ``Kokkos::View``) are detroyed
+before ``Kokkos::finalize`` gets called.
+
+Programs are ill-formed if they do not call this function after calling `Kokkos::initialize <initialize.html#kokkosinitialize>`_.
+
+Interface
+============
```
Comment (dalg24) [inline docs/source/API/core/initialize_finalize/finalize.rst:27]:
```suggestion
Interface
=========
```

**Comment by fnrizzi** [comment]
And the other reason is that we said this way we have a single consistent way of doing things

**Review by dalg24 — APPROVED** [review]

**Comment by fnrizzi** [comment]
@pierrepebay one small comment: for consistency, please use the same symbol to identify sections, subsections ,etc as in the other PRs see this for example: https://github.com/kokkos/kokkos-core-wiki/pull/204/files

**Review by fnrizzi — APPROVED** [review]

**Pull request merged**